src/main.rs

```diff
@@ -337,58 +382,55 @@ fn process_file(
         first_line = false;
 
         // that's a bad workaround
-        if line.trim() == "#[rustfmt::skip]" {
+        // Why do we require this, we dont use it (?)
+        if trimmed == "#[rustfmt::skip]" {
+            println!("Skipping rustfmt");
             continue;
         }
 
-        if line.trim().starts_with("#REPLACE ") {
-            let what = line.trim().strip_prefix("#REPLACE ").unwrap();
-            let mut split = what.split_terminator(" ");
-            replace = Some(split.next().unwrap().to_string());
-            let var = split.next().unwrap().to_string();
-            for (key, value) in variables {
-                if key == &var {
-                    replacement = Some(value);
-                    break;
-                }
-            }
-        } else if line.trim().starts_with("//REPLACE ") {
-            let what = line.trim().strip_prefix("//REPLACE ").unwrap();
+        // Check if we should replace the next line with the key/value of a variable
+        if let Some(what) = trimmed
+            .strip_prefix("#REPLACE ")
+            .or_else(|| trimmed.strip_prefix("//REPLACE "))
+        {
             let mut split = what.split_terminator(" ");
             replace = Some(split.next().unwrap().to_string());
             let var = split.next().unwrap().to_string();
-            for (key, value) in variables {
-                if key == &var {
-                    replacement = Some(value);
-                    break;
-                }
+
+            // Find the replacement value from the variables map
+            if let Some((_, value)) = variables.iter().find(|(key, _)| key == &var) {
+                replacement = Some(value);
             }
-        } else if line.trim().starts_with("#IF ") {
-            let cond = line.trim().strip_prefix("#IF ").unwrap();
-            if let Some(stripped) = cond.strip_prefix("!") {
-                include.push(!options.contains(&stripped.to_string()) && *include.last().unwrap());
+        // Check if we should include the next line(s)
+        } else if trimmed.starts_with("#IF ") || trimmed.starts_with("//IF ") {
+            let cond = if trimmed.starts_with("#IF ") {
+                trimmed.strip_prefix("#IF ").unwrap()
             } else {
-                include.push(options.contains(&cond.to_string()) && *include.last().unwrap());
-            }
-        } else if line.trim().starts_with("#ENDIF") {
-            include.pop();
-        } else if line.trim().starts_with("//IF ") {
-            let cond = line.trim().strip_prefix("//IF ").unwrap();
-            if let Some(stripped) = cond.strip_prefix("!") {
-                include.push(!options.contains(&stripped.to_string()) && *include.last().unwrap());
+                trimmed.strip_prefix("//IF ").unwrap()
+            };
+
+            scope.push_dynamic("cond", cond.into());
+
+            let result = engine
```

Comment:
my thinking was we could just pass the whole condition to Rhai and evaluate that

then we could make `contains_option` just operate on one option and we can write conditions like `contains_option("wifi") && (contains_option("alloc) || contains_option("embassy"))`